machine-learning/ml_tips/queries/cardinality.sql

```diff
@@ -0,0 +1,6 @@
+-- client: molehill/0.0.1
+select
+  array_max(array[approx_distinct(embarked), approx_distinct(sex), approx_distinct(pclass)]) as max_categorical_cardinality
```

Comment:
Can you add some comments about the purpose of this query? Maybe around `+compute_cardinality:` in the workflow definition? 

Auto-generated query is great, but it doesn't mean we can drop all comments... When we first see the queries, users might have no idea what it does, even if they're experienced data scientist; query-based ML is completely different from traditional ways of implementing ML e.g., Python code.

Reply:
Good catch.

This is the workaround to avoid this issue https://issues.apache.org/jira/browse/HIVEMALL-243
Without this workaround, training takes too long time 15 mins for titanic data while it will be done in a minute with the workaround.

Anyway, this workaround should be deleted after v0.6.0 release, so will add the comment for it.

Reply:
Add note 997d035